lib/node_modules/@stdlib/math/stats/anova1/lib/make_anova.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+// MODULES //
+
+var isNumberArray = require( '@stdlib/assert/is-number-array' ).primitives;
+var isTypedArrayLike = require( '@stdlib/assert/is-typed-array-like' );
+var isArray = require( '@stdlib/assert/is-array' );
+var setReadOnly = require( '@stdlib/utils/define-read-only-property' );
+var cdf = require( '@stdlib/math/base/dists/f/cdf' );
+var copy = require( '@stdlib/utils/copy' );
+var defaults = require( './defaults.json' );
+var validate = require( './validate.js' );
+var uniqueElems = require('./unique_elems.js');
+var meanTable = require('./mean_table.js');
+var prettyPrint = require( './print.js' );
+
+
+/**
+* Make the ANOVA Object.
+*
+* @param {NumericArray} x - Measured values
+* @param {Array} factor - Array of treatments
+* @param {Options} [options] - function options
+* @param {number} [options.alpha=0.05] - significance level
+* @param {boolean} [options.decision=false] - Indicate whether we reject the null hypothesis
+* @returns {Object} Returns the ANOVA Structure
+*/
+function anova1(x, factor) {
+	var err;
+	var means;
+	var treats;
+	var grandMean;
+	var i;
+	var sumSqTotal;
+	var sq; // Inside the double summation
+	var ssTreatment;
+	var treatment; // Index variable
+	var sumSqError;
+	var meanSumSqTreat; // Mean sum of squares
+	var meanSumSqError;
+	var fScore;
+	var numDf;
+	var denDf;
+	var pVal;
+	var out;
+	var opts;
+
+	// Initialize all vars
+	grandMean = 0;
+	sumSqTotal = 0;
+	ssTreatment = 0;
+
+	opts = copy( defaults );
+
+	// First check pre-conditions
+	if ( !isTypedArrayLike( x ) && !isNumberArray( x )) {
+		throw new TypeError( 'invalid first argument. x must be numeric array. Value: `' + x + '`.');
+	}
+
+	// Validate the options
+	if (arguments.length > 2) {
+		err = validate(opts, arguments[2]);
+		if ( err ) {
+			throw err;
+		}
+	}
+
+	if (x.length <= 1) {
+		throw new RangeError( 'length of x must be greater than or equal to 2.');
+	}
+
+	// Check if factor is an array
+	if ( !isArray( factor )) {
+		throw new TypeError(' factor argument must be an array' );
+	}
+
+	if (factor.length <= 1) {
+		throw new RangeError('Second argument must have at least 2 elements');
```

Comment:
I think we can remove this if-clause since we check in L83 for two unique elements.

Reply:
I updated the RangeError text and removed the L83 conditional, as I am not quite sure what I wanted to accomplish with the former. The latter checks for at least two unique elements, L83 was a relic from when the function could return null.